sky/cli.py

```diff
@@ -2983,6 +2983,32 @@ def _output():
                                                    clouds=cloud,
                                                    case_sensitive=False,
                                                    all_regions=all_regions)
+        # Import here to save module load speed.
+        from sky.clouds.service_catalog import common  # pylint: disable=import-outside-toplevel,line-too-long
+        # For each gpu name (count not included):
+        #   - Group by cloud
+        #   - Sort within each group by prices
+        #   - Sort groups by each cloud's (min price, min spot price)
+        new_result = {}
```

Comment:
Nice! It is great that we can have additional sorting ability for our `sky show-gpus`. A long existing issue I was wondering was that if we can group by the `(GPU, gpu_count)` and sort by price, instead of group by cloud. 
Most of the time I looked at the table, I tried to directly compare those machines with the same config across clouds, and our current way mixes different count makes it quite hard to compare. Wdyt?

Reply:
IIRC the reason we did it this way was to optimize for single-cloud users. For them, it'd be messy to group by (GPU, gpu_count, cloud) as they need to skim across many irrelevant clouds.

Reply:
For a single cloud user, they can always specify `--cloud` in `show-gpus` to get the results for that cloud only, so it might not be a blocker for the change.

I think we didn't do the `(GPU, gpu_count, cloud)` because each cloud groups the information retrieved from the catalog and we did not have the regrouping and resorting code path in `CLI` before. Since we have the "frontend" sorting enabled, we could probably switch to the tuple.
IMO, in most cases, comparing different configs of GPUs may not be as useful as comparing the same config but with different pricing. 

That said, I am fine with having the current version merged first and have another issue for the proposed change for the different key for grouping.

Reply:
I think we want to optimize for single-cloud only users. For them it's cumbersome to type an additional --cloud argument because it should be inferred as unique. Given these, I believe the current treatment is good.

Reply:
OK, I think I can now think of a benefit for grouping by cloud:
It will be easier to see all the available number of GPUs offered by a specific cloud.

While for the proposed key for grouping, it will be easier to find which cloud offers the specific number of GPU requested.

I am still half-half for this though, since most of the time a user knows how many GPUs they want for their workloads, e.g., 7B model requires a single L4, while 70B requires at least 4 40GB GPUs.
It should be fine to merge this PR first, as it is definitely better than the current unsorted version.

Reply:
> It will be easier to see all the available number of GPUs offered by a specific cloud.

Yep. Also, if one's single-cloud only, easier to show that cloud's offerings in a consecutive group.

> While for the proposed key for grouping, it will be easier to find which cloud offers the specific number of GPU requested.

For that, if users pass the count in the argument, both ways of grouping are the same (for most cases, except some where the same cloud has different X:1 offerings).